packages/codemirror/src/editor.ts

```diff
@@ -1080,19 +1089,32 @@ namespace CodeMirrorEditor {
  * The namespace for module private data.
  */
 namespace Private {
-  /**
-   * Handle the codemirror configuration options.
-   */
   export
-  function handleConfig(editor: CodeMirror.Editor, config: Partial<CodeMirrorEditor.IConfig>): void {
-    let fullConfig: CodeMirrorEditor.IConfig = {
-      ...CodeMirrorEditor.defaultConfig,
-      ...config
+  function createEditor(host: HTMLElement, config: CodeMirrorEditor.IConfig): CodeMirror.Editor {
+    let {
+      autoClosingBrackets,
+      fontFamily,
+      fontSize,
+      insertSpaces,
+      lineWrap,
+      tabSize,
+      readOnly,
+      ...otherOptions
+    } = config;
```

Comment:
Having to enumerate all of these seems like it will be easy to miss when something changes... perhaps explore going back to `Partial`...

Reply:
I agree that this is easy to miss things if something changes. I don't think it's *too* bad though. 

I thought it might be nice to annotate `bareConfig` as a `CodeMirror.EditorConfiguration` object so that the typescript compiler could catch some of that stuff.  However, when I tried that I found that the CodeMirror typings were not quite up to snuff. That being the case, I think this is a fine solution.